File: game/phone_events.py
from dataclasses import dataclass


@dataclass
class PhoneEvents:
    offhook: bool = False
    hungup: bool = False
 
    press_1: bool = False   
    press_2: bool = False
    press_3: bool = False
    press_4: bool = False
    press_5: bool = False
    press_6: bool = False
    press_7: bool = False
    press_8: bool = False
    press_9: bool = False
    press_0: bool = False
    press_star: bool = False
    press_pound: bool = False

    def any_set(self):
        return any([
            self.offhook,
            self.hungup,
            self.press_1,
            self.press_2,
            self.press_3,
            self.press_4,
            self.press_5,
            self.press_6,
            self.press_7,
            self.press_8,
            self.press_9,
            self.press_0,
            self.press_star,
            self.press_pound,
        ])

File: game-web/src/phoneEvents.ts
export class PhoneEvents {
  offhook = false;
  hungup = false;
  press_0 = false;
  press_1 = false;
  press_2 = false;
  press_3 = false;
  press_4 = false;
  press_5 = false;
  press_6 = false;
  press_7 = false;
  press_8 = false;
  press_9 = false;
  press_star = false;
  press_pound = false;

  anySet(): boolean {
    return (
      this.offhook || this.hungup ||
      this.press_0 || this.press_1 || this.press_2 || this.press_3 ||
      this.press_4 || this.press_5 || this.press_6 || this.press_7 ||
      this.press_8 || this.press_9 || this.press_star || this.press_pound
    );
  }
}
